### PR TITLE
chore: do not enable debugpy in production mode

### PR DESCRIPTION
**Pull request by bruno-fs**
- get rid of more docker references
- don't enable debugpy in "production" mode
- add QUIPUCORDS_ prefix to PRODUCTION env var

### CODE REVIEW

**Review by infinitewarp — APPROVED** [review]

File: quipucords/utils/debugger.py [hunk 1]
```diff
@@ -10,11 +10,14 @@
 
 DEBUGPY = env.bool("QUIPUCORDS_DEBUGPY", False)
 DEBUGPY_PORT = env.int("QUIPUCORDS_DEBUGPY_PORT", 5678)
+PRODUCTION = env.bool("PRODUCTION", False)
```
Comment (bruno-fs) [inline quipucords/utils/debugger.py:13]: change to QUIPUCORDS_PRODUCTION and the default to True

**Comment by infinitewarp** [comment]
Very strange. I don't understand why the test run failed these:
```
ERROR quipucords/tests/scanner/network/processing/test_hostnamectl.py - AttributeError: 'Settings' object has no attribute 'TESTDATA_DIR'
ERROR quipucords/tests/scanner/network/processing/test_hostnamectl.py - AttributeError: 'Settings' object has no attribute 'TESTDATA_DIR'
```
I don't see anything in the diff that should have affected these. 🤔 Maybe try force rerunning the test job? Do you see this error when running tests locally, @bruno-fs ?

**Comment by bruno-fs** [comment]
>Do you see this error when running tests locally, @bruno-fs ?
I don't. I was very puzzled to see these errors.

retrying now

**Comment by infinitewarp** [comment]
I reproduced locally by switching to your branch and running
```
poetry run pytest quipucords/tests/scanner/network/processing/test_hostnamectl.py
```

```
================================================================================ ERRORS =================================================================================
___________________________________________ ERROR collecting quipucords/tests/scanner/network/processing/test_hostnamectl.py ____________________________________________
quipucords/tests/scanner/network/processing/test_hostnamectl.py:13: in <module>
    hostnamectl_files_dir = settings.TESTDATA_DIR / "hostnamectl-status"
../../Library/Caches/pypoetry/virtualenvs/quipucords-A54Z8Zh8-py3.12/lib/python3.12/site-packages/django/conf/__init__.py:104: in __getattr__
    val = getattr(_wrapped, name)
E   AttributeError: 'Settings' object has no attribute 'TESTDATA_DIR'
=========================================================================== warnings summary ============================================================================
../../Library/Caches/pypoetry/virtualenvs/quipucords-A54Z8Zh8-py3.12/lib/python3.12/site-packages/environ/compat.py:14
  /Users/brasmith/Library/Caches/pypoetry/virtualenvs/quipucords-A54Z8Zh8-py3.12/lib/python3.12/site-packages/environ/compat.py:14: DeprecationWarning: 'pkgutil.find_loader' is deprecated and slated for removal in Python 3.14; use importlib.util.find_spec() instead
    if find_loader('simplejson'):

../../Library/Caches/pypoetry/virtualenvs/quipucords-A54Z8Zh8-py3.12/lib/python3.12/site-packages/environ/compat.py:19
  /Users/brasmith/Library/Caches/pypoetry/virtualenvs/quipucords-A54Z8Zh8-py3.12/lib/python3.12/site-packages/environ/compat.py:19: DeprecationWarning: 'pkgutil.find_loader' is deprecated and slated for removal in Python 3.14; use importlib.util.find_spec() instead
    if find_loader('django'):

../../Library/Caches/pypoetry/virtualenvs/quipucords-A54Z8Zh8-py3.12/lib/python3.12/site-packages/environ/compat.py:52
  /Users/brasmith/Library/Caches/pypoetry/virtualenvs/quipucords-A54Z8Zh8-py3.12/lib/python3.12/site-packages/environ/compat.py:52: DeprecationWarning: 'pkgutil.find_loader' is deprecated and slated for removal in Python 3.14; use importlib.util.find_spec() instead
    if old_django or not find_loader('pymemcache'):

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
======================================================================== short test summary info ========================================================================
ERROR quipucords/tests/scanner/network/processing/test_hostnamectl.py - AttributeError: 'Settings' object has no attribute 'TESTDATA_DIR'
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```
😕

**Comment by infinitewarp** [comment]
Flipping `PRODUCTION` default back to `False` appears to "fix" this, but I don't understand why.

```diff
diff --git a/quipucords/quipucords/settings.py b/quipucords/quipucords/settings.py
index 49d7bdd1..5cca7ef4 100644
--- a/quipucords/quipucords/settings.py
+++ b/quipucords/quipucords/settings.py
@@ -30,7 +30,7 @@ BASE_DIR = Path(__file__).absolute().parent.parent
 # DEFAULT_DATA_DIR is ./var, which is on .gitignore
 DEFAULT_DATA_DIR = Path(env.str("QUIPUCORDS_DATA_DIR", BASE_DIR.parent / "var"))

-PRODUCTION = env.bool("QUIPUCORDS_PRODUCTION", True)
+PRODUCTION = env.bool("QUIPUCORDS_PRODUCTION", False)
 SECURE_PROXY_SSL_HEADER = ("HTTP_X_FORWARDED_PROTO", "https")

 # This suppresses warnings for models where an explicit primary key is not defined.
```

**Comment by infinitewarp** [comment]
I may have found it:

```
if not PRODUCTION:
    TESTDATA_DIR = BASE_DIR / "testdata"
```

There are several things conditional on `PRODUCTION` that might break with the new default.

File: quipucords/quipucords/settings.py [hunk 2]
```diff
@@ -315,9 +315,6 @@ def app_secret_key_and_path():
 
 STATICFILES_DIRS = [BASE_DIR / "client"]
 
-if not PRODUCTION:
-    TESTDATA_DIR = BASE_DIR / "testdata"
```
Comment (bruno-fs) [inline quipucords/quipucords/settings.py:321]: testdata_dir could be a fixture. but I was too lazy right now to do it

**Pull request merged**